platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/InboundConnectionHandler.java

```diff
@@ -89,43 +80,24 @@ public InboundConnectionHandler(
      *
      * @param clientSocket the newly created socket
      */
-    public void handle(final Socket clientSocket) {
-        SerializableDataInputStream dis = null;
-        SerializableDataOutputStream dos = null;
-        NodeId otherId = null;
+    public void handle(final Socket clientSocket, final List<PeerInfo> peerInfoList) {
         long acceptTime = 0;
         try {
             acceptTime = System.currentTimeMillis();
             clientSocket.setTcpNoDelay(socketConfig.tcpNoDelay());
             clientSocket.setSoTimeout(socketConfig.timeoutSyncClientSocket());
 
+            PeerInfo connectedPeer = null;
+            if (clientSocket instanceof final SSLSocket sslSocket) {
+                connectedPeer = getConnectedPeer(sslSocket, peerInfoList);
+            }
+
             final SyncInputStream sis = SyncInputStream.createSyncInputStream(
                     platformContext, clientSocket.getInputStream(), socketConfig.bufferSize());
             final SyncOutputStream sos = SyncOutputStream.createSyncOutputStream(
                     platformContext, clientSocket.getOutputStream(), socketConfig.bufferSize());
 
-            dis = new SerializableDataInputStream(sis);
-            dos = new SerializableDataOutputStream(sos);
-
-            if (doVersionCheck) {
-                dos.writeSerializable(softwareVersion, true);
-                dos.flush();
-
-                final SoftwareVersion otherVersion = dis.readSerializable(Set.of(softwareVersion.getClassId()));
-                if (otherVersion == null
-                        || otherVersion.getClass() != softwareVersion.getClass()
-                        || otherVersion.compareTo(softwareVersion) != 0) {
-                    throw new IOException("This node has software version " + softwareVersion
-                            + " but the other node has software version " + otherVersion + ". Closing connection.");
-                }
-            }
-
-            final String otherKey = dis.readUTF();
-
-            otherId = addressBook.getNodeId(otherKey);
-
-            dos.writeInt(ByteConstants.COMM_CONNECT); // send an ACK for creating connection
-            dos.flush();
+            final NodeId otherId = Objects.requireNonNull(connectedPeer).nodeId();
```

Comment:
if the socket is not an SSLSocket this will throw an NPE, this is confusing, it should be handled differently. If we only support SSL sockets, then we should one use those

Reply:
I proposed getting rid of plain TCP as no one uses it today (as far as I can tell), and made this design assumption based on plain tcp connections going away (I have an unsubmitted PR for it).

If we want to support both, my preference will be to have two implementations of this class - one that figures out the other peer like before, and a new one that uses the new peer identification logic. The former will continue to require the address book to identify the connected peer, which re-introduces the separation of concern problem - one of the goals of isolating this network component. 

Another option of course is to throw a bunch of if else in this one class to support both. Not a fan but it may work. What do you think?

Reply:
Just a comment on code aesthetics.  Line 100 has 2 purposes, assert the invariant that the connectedPeer is not null and get the node id.   That first purpose feels like it should be closer to the definition of `connectedPeer`, like we wouldn't move forward into lines 95-98 if line 100 is going to fail as a result of the outcome of lines 95.    I think it feels easier to understand code when invariants are checked/enforced in proximity to the code which might violate them.

Reply:
`If we want to support both, my preference will be to have two implementations of this class`

An `InboundSSLConnectionHandler` and an `InboundNonSSLConnectionHandler`? 

If there is a lot of code duplication between the two classes, I'm in favor of it being a single class.   If the logic surrounding the if statements is large, duplicated between 2 (or more) classes, and needs to be updated, then we've created a scenario where code maintenance has to be made across all classes.    Keeping it one class with if statements minimizes the ancillary code duplication.    If there were special casing throughout the file for SSL vs NonSSL, then I'd say that might be a case for use of generic parameterized over the instance type.

Reply:
The epic says we don't have to support plain TCP, getting rid of it is easier

Reply:
thanks @edward-swirldslabs. I was thinking more like an AbstractInboundConnectionHandler that keeps the common code and two concretes that handle SSL vs Non-SSL. Thankfully we don't have to go this route. Makes all the sense to just get rid of it as @lpetrovic05 says.

Long term, I'll revisit this code to actually make the inbound and outbound handlers share code. Right now, they do quite similar things but can benefit from improved polymorphism.